ipn/ipnlocal/serve.go

```diff
@@ -247,7 +247,9 @@ func (b *LocalBackend) setServeConfigLocked(config *ipn.ServeConfig) error {
 	if err := b.store.WriteState(confKey, bs); err != nil {
 		return fmt.Errorf("writing ServeConfig to StateStore: %w", err)
 	}
-
+	for key := range config.Foreground {
+		mak.Set(&b.activeServeConfigForegrounds, key, struct{}{})
```

Comment:
wait, isn't this backwards? I thought the plan was to delete from this map for sessions that are no longer alive, where active is defined in terms of who's connected to the bus, not what's in the config.

As is, you're saying anything saved is considered alive and anything read stays undeleted if it is considered alive... which I guess kinda works out due to ordering: the first load is before the first save and nothing has been added to this set yet. But after that this set is kinda weird and makes me think this could just be a bool.

I'd prefer if we didn't mutate this here and instead kept this activeServeConfigForegrounds map set in sync based on connects and disconnects of bus watchers instead.

Reply:
Per our chat, we'll set/unset entries in this map from WatchIPNBus